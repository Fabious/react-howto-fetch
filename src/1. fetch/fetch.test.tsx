import Fetch from './fetch';
import renderer from 'react-test-renderer';

describe('Fetch', () => {
  let component: renderer.ReactTestRenderer;

  beforeEach(() => {
    // const mockResponseData = [{ id: 1 }, { id: 2 }, { id: 3 }];
    // global.fetch = jest.fn(async () => ({
    //   json: async () => mockResponseData,
    // }));

    renderer.act(() => {
      component = renderer.create(<Fetch />);
    });
  });

  it('fetches data with window.fetch', async () => {
    // await renderer.act(() => new Promise(() => {}));
    // component.update(<>pouet</>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
