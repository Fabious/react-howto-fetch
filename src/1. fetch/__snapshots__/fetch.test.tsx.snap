// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fetch fetches data with window.fetch 1`] = `
<p>
  there are 
  1
   users
</p>
`;
